src/test/java/teammates/storage/api/NotificationsDbTest.java

```diff
@@ -0,0 +1,229 @@
+        assertEquals(NotificationType.MAINTENANCE, notificationsDb.updateNotification(
+                NotificationAttributes.updateOptionsBuilder(n.getNotificationId())
+                        .withType(NotificationType.MAINTENANCE)
+                        .build()).getType());
+        assertEquals(NotificationType.MAINTENANCE, notificationsDb.getNotification(n.getNotificationId()).getType());
+
+        ______TS("success: single field - targetUser");
+        assertEquals(NotificationTargetUser.GENERAL, notificationsDb.getNotification(n.getNotificationId()).getTargetUser());
+        assertEquals(NotificationTargetUser.STUDENT, notificationsDb.updateNotification(
+                NotificationAttributes.updateOptionsBuilder(n.getNotificationId())
+                        .withTargetUser(NotificationTargetUser.STUDENT)
+                        .build()).getTargetUser());
+        assertEquals(NotificationTargetUser.STUDENT, notificationsDb.getNotification(n.getNotificationId()).getTargetUser());
+
+        ______TS("success: single field - startTime");
+        assertEquals(timeNow, notificationsDb.getNotification(n.getNotificationId()).getStartTime());
+        assertEquals(timeNow.plusSeconds(600), notificationsDb.updateNotification(
+                NotificationAttributes.updateOptionsBuilder(n.getNotificationId())
+                        .withStartTime(timeNow.plusSeconds(600))
+                        .build()).getStartTime());
+        assertEquals(timeNow.plusSeconds(600), notificationsDb.getNotification(n.getNotificationId()).getStartTime());
+
+        ______TS("success: single field - endTime");
+        assertEquals(timeNow.plusSeconds(3600), notificationsDb.getNotification(n.getNotificationId()).getEndTime());
+        assertEquals(timeNow.plusSeconds(4200), notificationsDb.updateNotification(
+                NotificationAttributes.updateOptionsBuilder(n.getNotificationId())
+                        .withEndTime(timeNow.plusSeconds(4200))
+                        .build()).getEndTime());
+        assertEquals(timeNow.plusSeconds(4200), notificationsDb.getNotification(n.getNotificationId()).getEndTime());
+
+        ______TS("success: single field - shown");
+        assertFalse(notificationsDb.getNotification(n.getNotificationId()).isShown());
+        assertTrue(notificationsDb.updateNotification(
+                NotificationAttributes.updateOptionsBuilder(n.getNotificationId())
+                        .withShown()
+                        .build()).isShown());
+        assertTrue(notificationsDb.getNotification(n.getNotificationId()).isShown());
+
+        ______TS("success: bulk update");
+        // We will try to revert back to the original attribute to test bulk update
+        // Note: Update of shown is not revertable so not to check it here
+        notificationsDb.updateNotification(
+                NotificationAttributes.updateOptionsBuilder(n.getNotificationId())
+                        .withTitle("title")
```

Comment:
Would it be better each build attribute into a variable at the start of the "success: bulk update" test so that we can easily see that the same attribute is used?

Reply:
Fixed in the latest commit.